gcc/rust/util/rust-attributes.cc

```diff
@@ -612,6 +613,22 @@ AttributeChecker::visit (AST::UseDeclaration &declaration)
   check_proc_macro_non_function (declaration.get_outer_attrs ());
 }
 
+static void
+check_no_mangle_function (const AST::Attribute &attribute,
+			  const AST::Function &fun)
+{
+  if (attribute.has_attr_input ())
+    {
+      rust_error_at (attribute.get_locus (), ErrorCode::E0754,
+		     "malformed %<no_mangle%> attribute input");
+      rust_inform (attribute.get_locus (),
+		   "must be of the form: %<#[no_mangle]%>");
+    }
+  if (!is_ascii_only (fun.get_function_name ().as_string ()))
+    rust_error_at (fun.get_function_name ().get_locus (),
+		   "the %<#[no_mangle]%> attribute requires ASCII identifier");
```

Comment:
```suggestion
		   "the %<#[no_mangle]%> attribute requires an ASCII identifier");
```

Reply:
Sorry for the late reply! I was busy last week.
This message comes from this actual rustc's message.
```
Compiling playground v0.0.1 (/playground)
error[[E0754]](https://doc.rust-lang.org/stable/error_codes/E0754.html): `#[no_mangle]` requires ASCII identifier
 --> src/main.rs:2:1
  |
2 | pub fn ああ() {}
  | ^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0754`.
error: could not compile `playground` (bin "playground") due to previous error
```

Reply:
oh, alright!